main.js

```diff
@@ -0,0 +1,181 @@
+var gpii = fluid.registerNamespace("gpii");
+const {app, Menu, Tray} = require('electron')
+var path = require("path");
+var request = require("request");
+var os = require("os");
+
+// The tray needs to be global
+var tray = null;
+
+// Component to start and stop parts of the GPII as well as key in and key out users
+fluid.defaults("gpii.app", {
+    gradeNames: "fluid.modelComponent",
+    model: {
+        gpiiStarted: false,
+        keyedInSet: null
+    },
+    listeners: {
+        onCreate: [{
+            funcName: "gpii.app.startLocalFlowManager"
+        }, {
+            funcName: "{that}.changeStarted",
+            args: [true]
+        }]
+    },
+    modelListeners: {
+        keyedInSet: [{ //  Will this be a race condition?
+            funcName: "gpii.app.keyOut",
+            args: ["{change}.oldValue"]
+        }, {
+            funcName: "gpii.app.keyIn",
+            args: ["{change}.value"]
+        }]
+    },
+    invokers: {
+        changeStarted: {
+            changePath: "gpiiStarted",
+            value: "{arguments}.0"
+        }
+    }
+});
+
+gpii.app.startLocalFlowManager = function () {
+    var fluid = require("universal"),
+        gpii = fluid.registerNamespace("gpii");
+
+    if (os.platform() === "win32") {
+        var windows = require("gpii-windows/index.js");
+    }
+
+    gpii.start();
+};
+
+gpii.app.keyIn = function (token) {
+    request("http://localhost:8081/user/"+token+"/login", function(error, response, body) {
+        //TODO Put in some error logging
+    });
+}
+
+gpii.app.keyOut = function (token) {
+    request("http://localhost:8081/user/"+token+"/logout", function(error, response, body) {
+        //TODO Put in some error logging
+    });
+}
+
+
+// Component to create and update the task tray menu.
+fluid.defaults("gpii.taskTray", {
+    gradeNames: "fluid.modelComponent",
+    model: { // Do I need to specify a default or will it sync up with the other component?
+        gpiiStarted: false,
+        keyedInSet: null
+    },
+    components: {
+        app: {
+            type: "gpii.app",
+            options: {
+                model: "{gpii.taskTray}.model"
+            }
+        }
+    },
+    listeners: {
+        onCreate: [{
+            funcName: "gpii.taskTray.makeTray",
+            args: ["{that}.options.icon"]
+        }, {
+            funcName: "gpii.taskTray.updateMenu",
+            args: ["{that}.model.gpiiStarted", "{that}.model.keyedInSet", "{that}.changeSet"]
+        }],
+    },
+    modelListeners: {
+        "*": {
+            funcName: "gpii.taskTray.updateMenu",
+            args: ["{that}.model.gpiiStarted", "{that}.model.keyedInSet", "{that}.changeSet"]
+        }
+    },
+    invokers: {
+        changeSet: {
+            changePath: "keyedInSet",
+            value: "{arguments}.0"
+        }
+    },
+    icon: "web/icons/gpii.ico"
+});
+
+gpii.taskTray.makeTray = function (icon) {
+    tray = new Tray(path.join(__dirname, icon));
```

Comment:
Tough that this object reference will be lost forever. Attach it to the component using an expander inside a member definition instead: https://github.com/fluid-project/infusion/blob/master/src/components/reorderer/js/Reorderer.js#L202